### PR TITLE
feat(linter): eslint-plugin-unicorn - no-empty-file

### PR DESCRIPTION
**Pull request by camc314**

### CODE REVIEW

File: crates/oxc_linter/src/rules/unicorn/no_empty_file.rs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    ///  - Empty statements
+    ///  - Empty blocks
+    ///  - Hashbang
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Meaningless files clutter a codebase.
+    ///
+    NoEmptyFile,
+    restriction,
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/unicorn/no_empty_file.rs:36]: Does useless files fit in the category of `correctness` 🤔

File: crates/oxc_linter/src/rules/unicorn/no_empty_file.rs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+);
+
+impl Rule for NoEmptyFile {
+    fn run_once(&self, ctx: &LintContext) {
+        println!("~~~~");
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/unicorn/no_empty_file.rs:41]: Oh no clippy is not catching this. Note to self: add this to clippy.

File: crates/oxc_linter/src/rules/unicorn/no_empty_file.rs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        let Some(root) = ctx.nodes().iter().next() else { return };
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/unicorn/no_empty_file.rs:42]: This is smart!

Comment (camc314) [reply crates/oxc_linter/src/rules/unicorn/no_empty_file.rs:41]: thanks! we could probably unwrap those two lines. but probably better to be safe 🙂

**Review by Boshen — APPROVED** [review]

**Pull request merged**